test/compilable/b16697.d

```diff
@@ -0,0 +1,12 @@
+version(D_SIMD)
+{
+    static assert(!is(         float       == __vector)); 
+    static assert(!is(         float[1]    == __vector)); 
+    static assert(!is(         float[4]    == __vector)); 
+    static assert( is(__vector(float[4])   == __vector)); 
+    static assert(!is(__vector(float[3])   == __vector)); 
+    static assert(!is(__vector(float[5])   == __vector)); 
+    static assert( is(__vector(float[4]) X == __vector) &&
```

Comment:
This is incorrect. See clause 6 under https://dlang.org/spec/expression.html#IsExpression
```
is(__vector(float[4]) X == __vector)
```
should set `X` to `float[4]`. This enables "peeling" off a layer of the type.